source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.cc

```diff
@@ -0,0 +1,130 @@
+}
+
+void QuicFilterManagerConnectionImpl::addBytesSentCallback(
+    Network::Connection::BytesSentCb /*cb*/) {
+  // TODO(danzh): implement to support proxy. This interface is only called from
+  // TCP proxy code.
+  ASSERT(false, "addBytesSentCallback is not implemented for QUIC");
+}
+
+void QuicFilterManagerConnectionImpl::enableHalfClose(bool enabled) {
+  ASSERT(!enabled, "Quic connection doesn't support half close.");
+}
+
+void QuicFilterManagerConnectionImpl::setBufferLimits(uint32_t /*limit*/) {
+  // TODO(danzh): add interface to quic for connection level buffer throttling.
+  // Currently read buffer is capped by connection level flow control. And
+  // write buffer is not capped.
+  ENVOY_CONN_LOG(error, "Quic manages its own buffer currently.", *this);
+}
+
+uint32_t QuicFilterManagerConnectionImpl::bufferLimit() const {
+  // As quic connection is not HTTP1.1, this method shouldn't be called by HCM.
+  NOT_REACHED_GCOVR_EXCL_LINE;
+}
+
+void QuicFilterManagerConnectionImpl::close(Network::ConnectionCloseType type) {
+  if (type != Network::ConnectionCloseType::NoFlush) {
+    // TODO(danzh): Implement FlushWrite and FlushWriteAndDelay mode.
+    ENVOY_CONN_LOG(error, "Flush write is not implemented for QUIC.", *this);
+  }
+  quic_connection_->CloseConnection(quic::QUIC_NO_ERROR, "Closed by application",
+                                    quic::ConnectionCloseBehavior::SEND_CONNECTION_CLOSE_PACKET);
+}
+
+void QuicFilterManagerConnectionImpl::setDelayedCloseTimeout(std::chrono::milliseconds timeout) {
+  ASSERT(timeout == std::chrono::milliseconds::zero(),
+         "Delayed close of connection is not supported");
+}
+
+std::chrono::milliseconds QuicFilterManagerConnectionImpl::delayedCloseTimeout() const {
+  // Not called outside of Network::ConnectionImpl. Maybe remove this interface
+  // from Network::Connection.
+  NOT_REACHED_GCOVR_EXCL_LINE;
+}
+
+const Network::ConnectionSocket::OptionsSharedPtr&
+QuicFilterManagerConnectionImpl::socketOptions() const {
+  ENVOY_CONN_LOG(
+      error,
+      "QUIC connection socket is merely a wrapper, and doesn't have any specific socket options.",
+      *this);
+  return quic_connection_->connectionSocket()->options();
+}
+
+const Network::Address::InstanceConstSharedPtr&
+QuicFilterManagerConnectionImpl::remoteAddress() const {
+  ASSERT(quic_connection_->connectionSocket() != nullptr,
+         "remoteAddress() should only be called after OnPacketHeader");
+  return quic_connection_->connectionSocket()->remoteAddress();
+}
+
+const Network::Address::InstanceConstSharedPtr&
+QuicFilterManagerConnectionImpl::localAddress() const {
+  ASSERT(quic_connection_->connectionSocket() != nullptr,
+         "localAddress() should only be called after OnPacketHeader");
+  return quic_connection_->connectionSocket()->localAddress();
+}
+
+Ssl::ConnectionInfoConstSharedPtr QuicFilterManagerConnectionImpl::ssl() const {
+  // TODO(danzh): construct Ssl::ConnectionInfo from crypto stream
```

Comment:
replace with NOT_IMPLEMENTED here and elsewhere?

Reply:
NOT_IMPLEMENTED will crash in integration test. But supporting handshake is not necessary for that.

Reply:
Per the comments above can we just remove the error logs? If something will be implemented later I think it's fine to just leave a TODO?

Essentially what I would like to see is:
1) No error logs in this type of code
2) NOT_IMPLEMENTED if it can't happen with a comment why.
3) Or a default implementation with a TODO to implement the real thing.

Reply:
Sorry I'm not following what is going on here. Presumably this does need to be implemented eventually but if you NOT_IMPLEMENTED this now it crashes? I would remove the error log and just track with a TODO.